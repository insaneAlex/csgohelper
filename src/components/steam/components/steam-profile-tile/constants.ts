export const LARGE_AVATAR_SIZE = 125;
export const AVATAR_SIZE = 75;
