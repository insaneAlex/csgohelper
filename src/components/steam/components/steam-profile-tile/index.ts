export {SteamProfileTile} from './steam-profile-tile';
