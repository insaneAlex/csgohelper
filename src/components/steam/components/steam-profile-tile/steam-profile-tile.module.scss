@import 'styles/index';

.profile {
  font-size: 15px;
  flex-direction: row;
  margin: 10px 0 10px 10px;

  @include box-container;
  @include large-screen {
    flex-direction: row;
    margin: 10px 10px 0;
  }

  img {
    margin-right: 10px;
    display: block;
  }

  .ownerName {
    margin-bottom: 25px;
  }

  .profileDescribe {
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    width: 150px;
    @include body-short;
  }
}
