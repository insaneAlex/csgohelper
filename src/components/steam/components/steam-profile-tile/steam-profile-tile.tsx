import {profileSelector} from '@/src/redux/features/inventory';
import {AVATAR_SIZE, LARGE_AVATAR_SIZE} from './constants';
import {useSelector} from 'react-redux';
import {useMedia} from '@/src/hooks';
import Image from 'next/image';
import Link from 'next/link';
import {FC} from 'react';

import styles from './steam-profile-tile.module.scss';

type Props = {itemsAmount: number; totalPrice: string};

export const SteamProfileTile: FC<Props> = ({itemsAmount, totalPrice}) => {
  const profile = useSelector(profileSelector);
  const isDesktop = useMedia({large: true});
  if (!profile) {
    return null;
  }
  const avatarSize = isDesktop ? LARGE_AVATAR_SIZE : AVATAR_SIZE;
  const {avatarfull, personaname, profileurl} = profile;

  return (
    <section className={styles.profile}>
      <div className={styles.imageWrapper}>
        <Link href={profileurl} target="_blank">
          <Image src={avatarfull} width={avatarSize} height={avatarSize} quality={50} priority alt="avatar" />
        </Link>
      </div>

      <div className={styles.name}>
        <p>{personaname}&#39;s inventory:</p>
        <div>
          <p>Items: {itemsAmount}</p>
          {totalPrice && <p>Value: {totalPrice}$</p>}
        </div>
      </div>
    </section>
  );
};
