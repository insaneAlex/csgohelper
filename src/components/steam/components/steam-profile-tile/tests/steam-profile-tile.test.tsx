import {SteamProfileTile} from '../steam-profile-tile';
import {render} from '@testing-library/react';

const useSelectorMock = jest.fn();
jest.mock('react-redux', () => ({useSelector: () => useSelectorMock(), useDispatch: () => jest.fn()}));

const profileMock = {avatarfull: 'https://avatars.full.jpg', personaname: 'boobaii', profileurl: 'url'};
describe('SteamProfileTile', () => {
  it('should render correctly', () => {
    useSelectorMock.mockReturnValue(profileMock);
    const {container} = render(<SteamProfileTile itemsAmount={111} totalPrice="111$" />);
    expect(container).toMatchSnapshot();
  });
});
