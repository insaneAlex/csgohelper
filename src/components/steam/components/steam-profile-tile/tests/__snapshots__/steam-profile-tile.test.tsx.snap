// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SteamProfileTile should render correctly 1`] = `
<div>
  <section
    class="profile"
  >
    <a
      href="url"
      target="_blank"
    >
      <img
        alt="avatar"
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height="80"
        src="/_next/image?url=https%3A%2F%2Fava.jpg&w=256&q=50"
        srcset="/_next/image?url=https%3A%2F%2Fava.jpg&w=96&q=50 1x, /_next/image?url=https%3A%2F%2Fava.jpg&w=256&q=50 2x"
        style="color: transparent;"
        width="80"
      />
    </a>
    <div
      class="profileDescribe"
    >
      <p
        class="ownerName"
      >
        boob
      </p>
      <p>
        Items: 
        111
      </p>
      <p>
        Value: 
        111$
        $
      </p>
    </div>
  </section>
</div>
`;
